addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -855,20 +865,36 @@ function ThreadFactory({ Entity }) {
          * @override
          */
         _updateAfter(previous) {
-            if (
-                this.model === 'mail.channel' &&
-                previous.foldState &&
-                this.foldState !== previous.foldState
-            ) {
-                this.notifyFoldStateToServer();
+            if (this.model === 'mail.channel') {
+                if (
+                    this.foldStatePending &&
+                    previous.foldStatePending !== this.foldStatePending
+                ) {
+                    this.notifyFoldStateToServer();
+                }
+                if (
+                    this.foldStateServer === this.foldStatePending
+                ) {
+                    this.update({ foldStatePending: undefined });
+                }
```

Comment:
I think both could be turned to compute methods. `notifyFoldStateToServer()` is done when `pendingFoldState` changes and `pendingFoldState !== serverFoldState`. This compute methods could be invoked from an attribute field e.g. `lastNotifiedFoldState`, and it shouldn't notify same fold state twice in a row. `pendingFoldState` could also be computed from `serverFoldState` changes and auto-reset when matching `serverFoldState`.

Reply:
This is probably correct but I'm afraid to change due to how little this is tested, and I'm not sure if we really have time to both write test and make this refactoring now.

Reply:
Ok, sounds reasonable to make this change later.

Reply:
task-2273658